lib/jsmidgen.js

```diff
@@ -393,8 +393,15 @@ var Midi = {};
 		var trackCount = this.tracks.length.toString(16);
 
 		// prepare the file header
-		var bytes = File.HDR_CHUNKID + File.HDR_CHUNK_SIZE + File.HDR_TYPE0;
+		var bytes = File.HDR_CHUNKID + File.HDR_CHUNK_SIZE;
 
+		// set Midi type based on number of tracks
+		if (parseInt(trackCount) > 1) {
```

Comment:
Could we just use `this.tracks.length` here instead, to avoid converting to/from hex?

That being said, I'm not sure whether Type 0 support is entirely necessary. For example, it is valid for a Type 1 MIDI file to have just a single track, whereas Type 0 **must** only have a single track. Perhaps we should allow the user to specify that they really want a Type 0 file, and default to Type 1.

When it comes to Type 2, I don't think there is much support for them (judging by http://www.midi.org/aboutmidi/tut_midifiles.php which is cited on Wikipedia), so I'm fine with not supporting them even if Type 0 support is added.